File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    left, sum, minLength := 0, 0, len(nums) + 1

    for right := 0; right < len(nums); right++ {
        sum += nums[right]

        for sum >= target {
            minLength = min(right - left + 1, minLength)
            sum -= nums[left]
            left++
        }
    }

    if minLength == len(nums) + 1 {
        return 0
    }

    return minLength
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.php
class Solution {

    /**
     * @param Integer $target
     * @param Integer[] $nums
     * @return Integer
     */
    function minSubArrayLen($target, $nums) {
        $sum = 0; $left = 0; $minLength = count($nums) + 1;

        for ($right = 0; $right < count($nums); $right++) {
            $sum += $nums[$right];

            while ($sum >= $target) {
                $minLength = min($minLength, $right - $left + 1);
                $sum -= $nums[$left];
                $left++;
            }
        }

        return $minLength === count($nums) + 1 ? 0 : $minLength;
    }
}